profiler/src/test/java/com/splunk/opentelemetry/profiler/snapshot/SpecialTraceIds.java

```diff
@@ -0,0 +1,256 @@
+
+  private static Map.Entry<Integer, List<String>> mapEntry(int percentile, String... traceIds) {
+    return Map.entry(percentile, List.of(traceIds));
+  }
+
+  /**
+   * Generates a map of percentiles to trace IDs that will pass the sampling check in {@link
+   * io.opentelemetry.sdk.trace.samplers.TraceIdRatioBasedSampler} when the corresponding
+   * probability is used (e.g., 5th percentile and 0.05).
+   */
+  public static void main(String[] args) {
+    var ranges = createRanges();
+    var traceIds = generateTraceIds(ranges);
+    System.out.println(asJavaCode(traceIds));
+  }
+
+  private static Queue<PercentileRange> createRanges() {
+    var ranges = new ArrayDeque<PercentileRange>();
+    for (int percentile = 1; percentile <= 99; percentile++) {
+      ranges.add(new PercentileRange(percentile, false));
+      ranges.add(new PercentileRange(percentile, true));
+    }
+    return ranges;
+  }
+
+  private static Map<Integer, List<String>> generateTraceIds(Queue<PercentileRange> ranges) {
+    var map = new LinkedHashMap<Integer, List<String>>();
+    while (!ranges.isEmpty()) {
+      var range = ranges.remove();
+      boolean foundTraceIdInRange = false;
+      while (!foundTraceIdInRange) {
+        var lower = new Random().nextLong(range.lower + 1, range.upper);
+        var traceId = TraceId.fromLongs(range.upper, lower);
+        map.compute(
+            range.value,
+            (percentile, traceIds) -> {
+              if (traceIds == null) {
+                traceIds = new ArrayList<>();
+              }
+              traceIds.add(traceId);
+              return traceIds;
+            });
+        foundTraceIdInRange = true;
+      }
+    }
+    return map;
+  }
+
+  private static StringBuilder asJavaCode(Map<Integer, List<String>> map) {
+    var string = new StringBuilder();
+    string
+        .append("private static final Map<Integer, List<String>> TRACE_IDS =")
+        .append("\tMap.ofEntries(");
+    for (var entry : map.entrySet()) {
+      string.append("\t\tmapEntry(").append(entry.getKey()).append(", ");
+      for (var traceId : entry.getValue()) {
+        string.append("\"").append(traceId).append("\", ");
+      }
+      string.deleteCharAt(string.length() - 1);
+      string.deleteCharAt(string.length() - 1);
+      string.append("),");
+    }
+    string.deleteCharAt(string.length() - 1);
+    string.append(");");
+    return string;
+  }
+
+  private static class PercentileRange {
+    private final int value;
+    private final long lower;
+    private final long upper;
+
+    private PercentileRange(int value, boolean negative) {
+      this.value = value;
+      if (!negative) {
+        this.lower = (long) (Long.MAX_VALUE * ((double) (value - 1) / 100));
+        this.upper = (long) (Long.MAX_VALUE * ((double) (value) / 100));
+      } else {
+        this.upper = (long) (Long.MIN_VALUE * ((double) (value - 1) / 100));
+        this.lower = (long) (Long.MIN_VALUE * ((double) (value) / 100));
```

Comment:
since `Long.MAX_VALUE != -Long.MIN_VALUE` and the `TraceIdRatioBasedSampler` only uses `Long.MAX_VALUE` it is difficult to tell whether this recreates the same value range as the code in `TraceIdRatioBasedSampler`